staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -166,3 +166,47 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityAndFairnessConfiguration holds API Priority and Fairness parameters.
+type PriorityAndFairnessConfiguration struct {
+	metav1.TypeMeta
+
+	// WorkEstimatorConfig holds work estimator parameters.
+	WorkEstimatorConfiguration WorkEstimatorConfiguration
+}
+
+// WorkEstimatorConfiguration holds work estimator parameters.
+type WorkEstimatorConfiguration struct {
+	// ListWorkEstimatorConfiguration holds work estimator parameters related to list requests.
+	ListWorkEstimatorConfiguration ListWorkEstimatorConfiguration
+	// MutatingWorkEstimatorConfiguration holds work estimator
+	// parameters related to watches of mutating objects.
+	MutatingWorkEstimatorConfiguration MutatingWorkEstimatorConfiguration
+
+	// MaximumSeats is the maximum number of seats a request can occupy
+	//
+	// NOTE: work_estimate_seats_samples metric uses the value of maximumSeats
+	// as the upper bound, so when we change maximumSeats we should also
+	// update the buckets of the metric.
+	MaximumSeats uint64
+}
+
+// ListWorkEstimatorConfiguration holds work estimator parameters related to list requests.
+type ListWorkEstimatorConfiguration struct {
+	// ObjectsPerSeat is a number of objects from a single list request per seat.
+	ObjectsPerSeat float64
+}
+
+// MutatingWorkEstimatorConfiguration holds work estimator
+// parameters related to watches of mutating objects.
+type MutatingWorkEstimatorConfiguration struct {
+	// EventAdditionalDuration is the additional amount of time that a mutating request occupies seats
+	// to account for the work involved in sending the corresponding WATCH notifications.
+	EventAdditionalDuration metav1.Duration
```

Comment:
With the "enable" boolean gone but us still wanting a way to effectively disable, the way to effectively disable needs to be documented and actually effective.  There are two logical ways to do it.

- `EventAdditionalDuration=0` would seem a simple logical way to disable the extra behavior here, but IIRC the cod still plumps out the request width in this case.  We should either keep this (arguably surprising, arguably regular) behavior and document it, or document and implement the special handling for this case.
- `WatchesPerSeat=1e100` (or any other value larger than any possible number of watches) will disable the dallying to account for WATCH notifications.

Reply:
Changed the documentation to:
``` go
// MutatingWorkEstimatorConfiguration holds work estimator
// parameters related to watches of mutating objects.
type MutatingWorkEstimatorConfiguration struct {
	// EventAdditionalDuration is the additional amount of time that a mutating request occupies seats
	// to account for the work involved in sending the corresponding WATCH notifications. Setting its
	// value to 0 effectively disables this behavior but a request may still occupy seats.
	// +optional
	EventAdditionalDuration metav1.Duration
	// WatchesPerSeat is the ratio between the number of WATCHes that will get notified and
	// the number of seats occupied for EventAdditionalDuration to account for the work of
	// sending those notifications. Setting its value to 0 disables mutating work estimator.
	// Setting its value to a number higher than any number of watches disables this behavior but
	// a request may still occupy a seat as per EventAdditionalDuration.
	// +optional
	WatchesPerSeat float64
}
```

and added the following condition to the mutating work estimator:
``` go
if e.config.MutatingWorkEstimator.WatchesPerSeat == 0 {
	return WorkEstimate{
		InitialSeats: minimumSeats,
	}
}
```
Previously it didn't make much sense to set its value to 0 as it's used as a divider.